### PR TITLE
New BEP to add SHA3-256 hashing and ECRecoverPublic key Precompiled contracts to BSC

### PR DESCRIPTION
**Pull request by simsonraj**
This is a new draft BEP to add new hashing sha3-256 hashing functionalities to BSC

### CODE REVIEW

**Review by que1414 — APPROVED** [review]

**Comment by BennyOptions** [comment]
Hey @que1414 do you know when this PR could be merged and included in a hard-fork? We are trying to plan our deployment schedule for BSC and this PR is necessary for us, thanks!

**Comment by icon2infiniti** [comment]
@que1414 Can we get a response from you to see where we're at and if this can be merged? This is a critical dependency for us, without these functions merged, our project cannot be used in a real network setting. We have tested these on a custom built BSC node and they seem to be working just fine. This is also being deployed to other protocols in a similar fashion, and most have gone smoothly.

**Comment by BennyOptions** [comment]
Hey @que1414 hope you're doing well! Just wanted to follow up on this PR to see if this can be included in the next hard fork and also when the next hard fork is planned. Thanks!

**Comment by dab1999** [comment]
> This is a new draft BEP to add new hashing sha3-256 hashing functionalities to BSC

BEP2.md

**Comment by BennyOptions** [comment]
@HaoyangLiu @huangsuyu @ackratos @guagualvcha @que1414 

Hey guys - any chance somebody merge this PR and let us know when the next hard fork is coming? Seems it's already been reviewed and good to go, it's an important feature for our upcoming multi-chain bridge to connect BSC to several other blockchain ecosystems

File: BEP92.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+## 3. Status
+
+This BEP is a Work in Progress(WIP). 
+
+## 4. Motivation
+
+To bring more interoperability to Binance Smart Chain, this BEP proposing to add additional capabilities to verify sha3-256 based blockchains in the likes of ICON republic.
+
+- In order to validate the merkle tree data from ICON and other similar blockchains, which uses the SHA3-256 hashing function.
```
Comment (guagualvcha) [inline BEP92.md:27]: Please use `ICON` as less as possible and make the bep more generic. Thanks.

File: BEP92.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+To bring more interoperability to Binance Smart Chain, this BEP proposing to add additional capabilities to verify sha3-256 based blockchains in the likes of ICON republic.
+
+- In order to validate the merkle tree data from ICON and other similar blockchains, which uses the SHA3-256 hashing function. 
+- When the transaction hashes are created using the SHA3-256 hashing function and the same hashing mechanism is needed to validate signatures of validators in ICON. 
+- Currently ICON uses a similar address building scheme to ethereum. ICON uses a part of the hash value of the public key, but with with SHA3-256 FIPS hashing function. Hence the need for new ecrecoverPublicKey function as part of precompiled.
+
+## 5. Specification
+Ethereum based virtual machines uses Keccak-256 hashing algorithm to evaulate blocks, but most of the modern blockchains adapted SHA3-256,(slightly different padding rule) as part of [NIST FIPS 202](#https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.202.pdf).
+
```
Comment (guagualvcha) [inline BEP92.md:33]: Shall we introduce the gas consumption  of these two precompile contracts?

File: BEP92.md (hunk 1 above)
Comment (simsonraj) [reply BEP92.md:27]: @guagualvcha , Sure, I'll modify this.

File: BEP92.md (hunk 2 above)
Comment (simsonraj) [reply BEP92.md:33]: @guagualvcha , Ill add the gas details on this.

Comment (simsonraj) [reply BEP92.md:36]: Added, Please let me know if any changes needed on this, cc/ @que1414

Comment (guagualvcha) [reply BEP92.md:36]: Thanks.

File: BEP92.md (hunk 1 above)
Comment (guagualvcha) [reply BEP92.md]: thanks. LGTM

File: BEP92.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  - [7. Test Data](#7-testdata)
+  - [8. License](#6-license)
+  
+## 1. Summary
+
+This BEP describes a proposal for Implementation of SHA3-256 as precompiled for BSC
+
+## 2. Abstract
+
+BEP-92 Proposal describes a addition of implementation of SHA3-256 FIPS 202 hash and EcRecover Uncompressed public key methods as precompiled native contracts
```
Comment (guagualvcha) [inline BEP92.md:19]: `a addition` ==> `an addition`

Comment (guagualvcha) [reply BEP92.md:19]: Thanks @simsonraj,  the BEP-92 looks fine to me. No more comments

Comment (simsonraj) [reply BEP92.md]: Corrected this, thank you @guagualvcha!

**Review by ratthakorn2509 — APPROVED** [review]

**Review by quinio01 — COMMENTED** [review]
think it's good for my issue

**Review by Debu976116 — APPROVED** [review]

**Review by Debu976116 — APPROVED** [review]

**Review by Debu976116 — APPROVED** [review]

**Comment by brilliant-lx** [comment]
Did some research on this, due to the EVM-Compatible concern, it would be better to support SHA3 in Ethereum first and apply to all other EVM compatible chains.
If we support it in BSC with a precompile address, it could conflict if Ethereum finally support it with another precompile address.
As BSC, I am fine with it.
I know Ethereum community has rejected it in https://github.com/ethereum/EIPs/issues/2951, but it is closed mainly because it does not urgent and no one keep pushing it forward.

**Pull request closed without merging**